src/query-performance-monitoring/validator/validations.go

```diff
@@ -0,0 +1,248 @@
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			log.Error("Failed to close rows: %v", err)
+		}
+	}()
+
+	// Check if each essential instrument is enabled and timed
+	for rows.Next() {
+		var name, enabled string
+		var timed sql.NullString
+		if err := rows.Scan(&name, &enabled, &timed); err != nil {
+			return fmt.Errorf("failed to scan instrument row: %w", err)
+		}
+		if enabled != "YES" || (timed.Valid && timed.String != "YES") {
+			log.Error("Essential instrument %s is not fully enabled. To enable it, run: UPDATE performance_schema.setup_instruments SET ENABLED = 'YES', TIMED = 'YES' WHERE NAME = '%s';", name, name)
+			return fmt.Errorf("%w: %s", utils.ErrEssentialInstrumentNotEnabled, name)
+		}
+	}
+
+	if err := rows.Err(); err != nil {
+		return fmt.Errorf("rows iteration error: %w", err)
+	}
+
+	return nil
+}
+
+// logEnablePerformanceSchemaInstructions logs instructions to enable the Performance Schema.
+func logEnablePerformanceSchemaInstructions(db utils.DataSource) {
+	version, err := getMySQLVersion(db)
+	if err != nil {
+		log.Error("Failed to get MySQL version: %v", err)
+		utils.FatalIfErr(err)
+	}
+
+	if isVersion8OrGreater(version) {
+		log.Debug("To enable the Performance Schema, add the following lines to your MySQL configuration file (my.cnf or my.ini) in the [mysqld] section and restart the MySQL server:")
+		log.Debug("performance_schema=ON")
+
+		log.Debug("For MySQL 8.0 and higher, you may also need to set the following variables:")
+		log.Debug("performance_schema_instrument='%%=ON'")
+		log.Debug("performance_schema_consumer_events_statements_current=ON")
+		log.Debug("performance_schema_consumer_events_statements_history=ON")
+		log.Debug("performance_schema_consumer_events_statements_history_long=ON")
+		log.Debug("performance_schema_consumer_events_waits_current=ON")
+		log.Debug("performance_schema_consumer_events_waits_history=ON")
+		log.Debug("performance_schema_consumer_events_waits_history_long=ON")
+	} else {
+		log.Error("MySQL version %s is not supported. Only version 8.0+ is supported.", version)
+		utils.FatalIfErr(ErrMysqlVersion)
+	}
+}
+
+// getMySQLVersion retrieves the MySQL version from the database.
+func getMySQLVersion(db utils.DataSource) (string, error) {
+	query := "SELECT VERSION();"
+	rows, err := db.QueryX(query)
+	if err != nil {
+		return "", fmt.Errorf("failed to execute version query: %w", err)
+	}
+	defer rows.Close()
+
+	var version string
+	if rows.Next() {
+		if err := rows.Scan(&version); err != nil {
+			return "", fmt.Errorf("failed to scan version: %w", err)
+		}
+	}
+
+	if version == "" {
+		return "", utils.ErrMySQLVersion
+	}
+
+	return version, nil
+}
+
+// isVersion8OrGreater checks if the MySQL version is 8.0 or greater.
+func isVersion8OrGreater(version string) bool {
+	majorVersion, minorVersion := parseVersion(version)
+	return (majorVersion > 8) || (majorVersion == 8 && minorVersion >= 0)
```

Comment:
If we are sure this fuctionality is supported irrespective of the minorVersion? could this be simplified as below
```suggestion
	return (majorVersion >= 8)
```

Reply:
updated.